pkg/platform/kube/test/suite.go

```diff
@@ -198,25 +202,15 @@ func (suite *KubeTestSuite) GetResourceAndUnmarshal(resourceKind, resourceName s
 	suite.Require().NoError(err)
 }
 
-func (suite *KubeTestSuite) WaitForFunctionDeploymentAvailability(functionName string,
-	duration time.Duration) {
-
-	expectedUnavailableReplicas := 0
+func (suite *KubeTestSuite) WaitForFunctionDeployment(functionName string,
+	duration time.Duration,
+	callback func(*appsv1.Deployment) bool) {
 	err := common.RetryUntilSuccessful(duration,
 		time.Second,
 		func() bool {
-			deploymentInstance := &appsv1.Deployment{}
-			suite.GetResourceAndUnmarshal("deployment",
-				kube.DeploymentNameFromFunctionName(functionName),
-				deploymentInstance)
-			suite.Logger.InfoWith("Waiting for deployment unavailable replicas to be zero",
-				"currentUnavailableReplicas", deploymentInstance.Status.UnavailableReplicas,
-				"expectedUnavailableReplicas", expectedUnavailableReplicas)
-
-			// wait until all replicas are available
-			return int(deploymentInstance.Status.UnavailableReplicas) == expectedUnavailableReplicas
+			return callback(suite.getFunctionDeployment(functionName))
```

Comment:
I thought you didn't like the callback mess - why not just pass desired number of replicas (int)? for now you don't need anything else
KISS

Reply:
this function has one more usage than checking for replicas, much easier for suite and readable on test itself when using callback (in this scenario)

I won't say it is a mater of KISS or not, because what's important here is to provide easy way to add tests without needing to add more utils to suite itself.

Reply:
> "usage than checking for replicas,"
ok